### ISSUE TITLE
[Feature Request] Retro Achievements Support

### ISSUE THREAD

**Issue by ThisNekoGuy** [xemu-project/xemu#582]
### Feature Request

[Retro Achievements](https://retroachievements.org/) is something that's been implemented in a number of emulators including DuckStation / SwanStation, and many others - mostly maintained by [libretro*](https://retroachievements.org/download.php)

It would be great if Xemu had support for these as well; the Original Xbox had many great games and getting achievements for them would be a nice bonus :p

### Alternatives

- Not supporting achievements :(

### Additional Context

For reference in assisting this:
- [Retro Achievements Discord Server](https://discord.gg/dq2E4hE)
- [Retro Achievements Developer Docs](https://docs.retroachievements.org/How-to-Become-an-Achievement-Developer/)
- [Retro Achievements Github](https://github.com/RetroAchievements)

**Comment by mborgerson** [comment]
Thanks for the request. I don't intend for this to be a core feature of xemu, at least any time soon, but perhaps a plugin (or a script via qmp) can provide such functionality. I'll leave this issue open for tracking.

**Comment by Darknior** [comment]
It is really a good idea :) 
I don't remember if there were some real succes to unlock on XBOX ? On x360 yes there are !
But yes, it many challenge to players ... now all emulators are using this feature and it is excellent :) There are also challenge by month with it.
For sure our LOVED XBOX would have it too :)

**Comment by ThisNekoGuy** [comment]
Which reminds me: If Xemu ever gets a libretro core, having this would probably make implementing it on that side smoother too
Or maybe one of the libretro/RetroArch guys would be interested in working on it from the start :p

**Comment by Darknior** [comment]
Add Xemu to libretro will be a pain i think, maybe it will never append !
I will don't stand for it, and i love libretro, but many emulators port in libretro are slower too ...
Duckstation standalone has Retroachievement, we can do have them here too. It is not only a libretro feature.

**Comment by Maximdraco** [comment]
Multiple Retroachievement users are interested in Xemu, we would love to be able to get and create achievement for Xbox games.

**Comment by SK-Crow** [comment]
Just commenting that I would use this feature.  RA has brought me to playing a ton of old games all over again.

**Comment by Dettomi** [comment]
Retroachievements is a love letter for retro games/systems and a must have in my opinion. Looking at gen 6: Dreamcast, Playstation 2 is already implemented, Gamecube is pretty near completion (wii too, no libretro) so the logical next step would be our beloved console coming to the site. With an end of 2024, 2025 date it could be pretty doable and a big win (some would say ACHIEVEMENT) for hoomanity.

**Comment by toxyy** [comment]
Would love for Retroachievementrs to be implemented somehow.

**Comment by TGP482** [comment]
This could be a great way to encourage more people to download and use the emulator, especially considering the large community of achievement hunters. For instance, the Xemu Discord server has around 11,000 members, while the RetroAchievements server has roughly 18,000.

**Comment by ThisNekoGuy** [comment]
@TGP482 A more accurate statistic would be the RetroAchievements website, actually. Their Discord server accounts for a tiny fraction of the true userbase:
![Image](https://github.com/user-attachments/assets/73377123-91cb-49bc-b2a3-9c2d635b0294)

**Comment by RinMaru** [comment]
Does RetroAchievements even have xbox achievements? that kinda needs to happen before anything can happen.

**Comment by televandalist** [comment]
> Does RetroAchievements even have xbox achievements? that kinda needs to happen before anything can happen.

In order to make the achievements, we would need an Xbox emulator or core to support RAIntegration + rcheevos so our toolkit can be used with it.

After that, assuming the RAM is properly exposed and a hashing method is taken care of, we'd take the steps needed to start testing and do a rollout... which is when a bunch of our devs spend a few months knocking out sets so we can officially launch support with a decent number of sets ready to go.

**Comment by televandalist** [comment]
> This could be a great way to encourage more people to download and use the emulator, especially considering the large community of achievement hunters. For instance, the Xemu Discord server has around 11,000 members, while the RetroAchievements server has roughly 18,000.

On top of the increased traffic on both sides, there'd be about a hundred achievement developers doing pretty much everything that's possible to do in a ton of games, which is a win-win in regards to compatibility testing, finding and reporting bugs, etc. 

> I don't intend for this to be a core feature of xemu, at least any time soon, but perhaps a plugin (or a script via qmp) can provide such functionality. I'll leave this issue open for tracking.

Basically, [rcheevos](https://github.com/RetroAchievements/rcheevos) makes it possible for an emulator to process achievement data and for players to earn achievements. Some emulator developers have made their own rcheevos implementation, though. [RAIntegration](https://github.com/RetroAchievements/RAIntegration) makes it possible for our devs to use RA's toolkit with an emulator so we can make the achievements. This exists as a .dll that's typically added to the main directory. In DuckStation and PCSX2's case, it's a "hidden" feature that's enabled via manually editing the config or by using `-raintegration` as a command line argument when launching the emulator. So for the average player, those things are out of sight/out of mind.

**Comment by Tanedhelion** [comment]
I’ll make a comment here to voice my support as well.

**Comment by rubentavares** [comment]
I’m also voicing my support. Last year GameCube launched with achievements support via Dolphin and it was the biggest launch in the history of the site gaining a lot of traffic for both Dolphin and RA. 

Plus there are hundreds of super motivated achievement devs who would basically develop achievements for 100 games at launch easily.

**Comment by wescopeland** [comment]
The RetroAchievements emulator integration guide can be found here: https://github.com/RetroAchievements/rcheevos/wiki/rc_client-integration.

**Comment by KiriSweets** [comment]
Apparently we are treating this as a soft petition to voice our desire? Hopefully that's not bothersome somehow, but if not: I'm doing my part!

Would love to play Mechassault with Achievements. I only very briefly interacted with the original Xbox as a kid and RA achievement support would be a wonderful way to encourage myself to try more of the exclusives I've either never played or only barely touched. And also an excuse to replay JSRF again. You know that if Xemu got RA support someone would get that set up in like a day.

**Comment by mborgerson** [comment]
It's nice to see support in adding value to playing original Xbox games in xemu! To keep the thread substantive, feel free to leave a ❤  on the original issue post instead of a comment if you're just wanting to show interest.

There's still a lot of work to do in improving xemu generally, and development time is a limited resource, so it's difficult to say if or when this support might be added. If there's a Retro Achievements user or developer interested enough to work on support for it, ideally as a plugin as I mentioned above, they are welcome to do so. You are welcome to ask technical questions about doing so in the xemu Discord server.

**Comment by retrosoul86** [comment]
I think it would be awesome there are a ton of games that I would love to get cheevos for on the Xbox and have xemu myself

**Comment by Neo-Retro** [comment]
this is my first comment ever here on git, but Retroachievement support for xemu would be a real game changer for xbox emulation. 

I leave many time on retroachievements and can say we have so many developers he would develop on xbox classic games. 

i think personaly  on my favorit games Fable and Morrorwind, Retroachievements for that games would be so cool. at this moment i don't play xbox classic but with achievements i would play every free second.

please think about that feature, that would be very nice for both sides.
Retroachievement have so many retro gamer and a very big part from they player, play only games with achievement support. it's a great chance to make the XEMU community bigger.

**Comment by SlipSlot** [comment]
Xbox basically started platform-wide achievement support on consoles with the 360, the fact it has one of the few emulation projects that has little interest in integrating it is oddly sad for me.

Really hope this changes one day, Xbox of all consoles deserves it, the reason achievements even exist as they are today is basically thanks to Xbox.

**Comment by televandalist** [comment]
> Xbox basically started platform-wide achievement support on consoles with the 360, the fact it has one of the few emulation projects that has little interest in integrating it is oddly sad for me.
> 
> Really hope this changes one day, Xbox of all consoles deserves it, the reason achievements even exist as they are today is basically thanks to Xbox.

Yep. Dead Rising on 360 is pretty much what got me hooked and the official set for that is what I consider the gold standard for achievement lists.... even Transmissionary.

**Comment by mborgerson** [comment]
Locking this thread now due to continued duplicate comments petitioning for the feature.

As I mentioned above, the project's relatively limited development resources are being used to improve emulation and usability. I encourage those who really want original Xbox Retro Achievements on xemu to contribute some effort in making it a reality.
